### PR TITLE
Test & Refactor OpenPgpUtils

### PR DESCRIPTION
**Pull request by ligi**
I think we should talk about the case: Max Mustermann [this is a nothing] afterwards

### CODE REVIEW

File: k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+@RunWith(AndroidJUnit4.class)
+public class OpenPgpUtilsTest {
+
+    @Test
+    public void splitCompleteUserIdShouldReturnAll3Components() throws Exception {
+        OpenPgpUtils.UserInfo info = OpenPgpUtils.splitUserId("Max Mustermann (this is a comment) <max@example.com>");
+
+        assertEquals(info.name, "Max Mustermann");
```
Comment (jberkel) [inline k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java]: the `assertEquals` param order is `expected, actual`, so this (and all other assertions) should be written as
```java
assertEquals("Max Mustermann", info.name);
```

File: k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        assertEquals(info.email, "max@example.com");
+    }
+
+
+    @Test
+    public void splitUserIdWithAllButCommentShouldReturnNameAndEmail() throws Exception {
+        OpenPgpUtils.UserInfo info = OpenPgpUtils.splitUserId("Max Mustermann <max@example.com>");
+
+        assertEquals(info.name, "Max Mustermann");
+        assertEquals(info.comment, null);
```
Comment (jberkel) [inline k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java]: `assertNull(info.comment)`

File: k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java (hunk 1 above)
Comment (ligi) [inline k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java]: ack

File: k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java (hunk 2 above)
Comment (ligi) [inline k9mail/src/androidTest/java/com/fsck/k9/crypto/OpenPgpUtilsTest.java]: ack

**Comment by ligi** [comment]
@jberkel updated with your comments in mind - also handle the [this is nothing] case as agreed face2face with @dschuermann

**Pull request merged**